test/request.go

```diff
@@ -180,6 +181,7 @@ func WaitForEndpointState(
 // desc will be used to name the metric that is emitted to track how long it took for the
 // domain to get into the state checked by inState.  Commas in `desc` must be escaped.
 func WaitForEndpointStateWithTimeout(
+	ctx context.Context,
```

Comment:
Why do we plumb here but not in `WaitForDeploymentState` for example?

Reply:
done in 77678bb83f220dd422bf886e7ce009f6f23d6616